hat/assets/js/apps/Iaso/domains/dataSources/components/AddNewEmptyVersion.js

```diff
@@ -0,0 +1,121 @@
+import React from 'react';
+import { Grid, Typography } from '@material-ui/core';
+import { FormattedMessage } from 'react-intl';
+import { LoadingSpinner, useSafeIntl } from 'bluesquare-components';
+import PropTypes from 'prop-types';
+import MESSAGES from '../messages';
+import ConfirmCancelDialogComponent from '../../../components/dialogs/ConfirmCancelDialogComponent';
+import { createSourceVersion } from '../requests';
+import { useFormState } from '../../../hooks/form';
+import { useSnackMutation } from '../../../libs/apiHooks.ts';
+import InputComponent from '../../../components/forms/InputComponent';
+
+const initialFormState = () => {
+    return {
+        versionDescription: '',
+    };
+};
+
+const AddNewEmptyVersion = ({ renderTrigger, sourceId }) => {
+    const { formatMessage } = useSafeIntl();
+    // eslint-disable-next-line no-unused-vars
+    const [form, setFormField, _, setFormState] = useFormState(
+        initialFormState(),
+    );
+
+    const mutation = useSnackMutation(
+        createSourceVersion,
+        MESSAGES.newEmptyVersionSavedSuccess,
+        MESSAGES.newEmptyVersionError,
+    );
+
+    const reset = () => {
+        setFormState(initialFormState());
+    };
+
+    const submit = async closeDialogCallBack => {
+        const body = {
+            dataSourceId: sourceId,
+            description: form.versionDescription.value || null,
+        };
+        await mutation.mutateAsync(body);
+        closeDialogCallBack();
+        reset();
+    };
+
+    const onConfirm = async closeDialog => {
+        await submit(closeDialog);
+    };
+
+    const onRedirect = async closeDialog => {
+        await submit(closeDialog);
+    };
+
+    const titleMessage = (
+        <FormattedMessage
+            id="iaso.sourceVersion.label.createNewEmptyVersion"
+            defaultMessage="Create a new empty version"
+        />
+    );
+
+    const allowConfirm = !mutation.isLoading;
+
+    const renderDefaultLayout = versionNumber => {
+        return (
+            <>
+                {!versionNumber && (
+                    <InputComponent
+                        type="text"
+                        keyValue="versionDescription"
+                        labelString={formatMessage(
+                            MESSAGES.dataSourceDescription,
+                        )}
+                        value={form.versionDescription.value}
+                        onChange={(field, value) => {
+                            setFormField(field, value);
+                        }}
+                    />
+                )}
+            </>
+        );
+    };
+
+    return (
+        <ConfirmCancelDialogComponent
```

Comment:
Son did a new component in `bluesquare-components`, search `makeFullModal` ( for next time 😉 )

Reply:
Thank I will take in account for the next time.